dspace-api/src/main/java/org/dspace/content/RelationshipMetadataServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+        } else {
+            //No virtual metadata can be created
+            return resultingMetadataValueList;
+        }
+
+        if (hashMaps != null && enableVirtualMetadata) {
+            resultingMetadataValueList.addAll(handleRelationshipTypeMetadataMapping(context, item, hashMaps,
+                                                                                    otherItem, relationName,
+                                                                                    relationship, place, isLeftwards));
+        }
+        RelationshipMetadataValue relationMetadataFromOtherItem =
+            getRelationMetadataFromOtherItem(context, otherItem, relationName, relationship.getID(), place);
+        if (relationMetadataFromOtherItem != null) {
+            resultingMetadataValueList.add(relationMetadataFromOtherItem);
+        }
+        return resultingMetadataValueList;
+    }
+
+    //This method will retrieve a list of RelationshipMetadataValue objects based on the config passed along in the
+    //hashmaps parameter. The beans will be used to retrieve the values for the RelationshipMetadataValue objects
+    //and the keys of the hashmap will be used to construct the RelationshipMetadataValue object.
+    private List<RelationshipMetadataValue> handleRelationshipTypeMetadataMapping(Context context, Item item,
+        HashMap<String, VirtualMetadataConfiguration> hashMaps, Item otherItem, String relationName,
+        Relationship relationship, int place, boolean isLeftwards) throws SQLException {
+
+        List<RelationshipMetadataValue> resultingMetadataValueList = new LinkedList<>();
+        for (Map.Entry<String, VirtualMetadataConfiguration> entry : hashMaps.entrySet()) {
+            String key = entry.getKey();
+            VirtualMetadataConfiguration virtualBean = entry.getValue();
+
+            if (virtualBean.getPopulateWithNameVariant()) {
+                String wardLabel = isLeftwards ? relationship.getLeftwardValue() : relationship.getRightwardValue();
+                if (wardLabel != null) {
+                    resultingMetadataValueList.add(
+                        constructRelationshipMetadataValue(context, item, relationship.getID(), place, key, virtualBean,
+                                                           wardLabel));
+                } else {
+                    handleVirtualBeanValues(context, item, otherItem, relationship, place, resultingMetadataValueList,
+                                            key, virtualBean);
+                }
+            } else {
+                handleVirtualBeanValues(context, item, otherItem, relationship, place, resultingMetadataValueList, key,
+                                        virtualBean);
+            }
+        }
+        return resultingMetadataValueList;
+    }
+
+    private void handleVirtualBeanValues(Context context, Item item, Item otherItem, Relationship relationship,
```

Comment:
This `handle` method looks more like it is an `add` to an in memory list?  Perhaps it should be named `add` in some manner?

Reply:
I've renamed it to findVirtualMetadataFromConfiguration and refactored it to return `List<RelationshipMetadataValue>` instead of adding it to the resultingMetadataValueList parameter.
Combined with the JavaDocs I hope this clarifies the method's purpose